controller/controller.go

```diff
@@ -139,6 +146,52 @@ func (c *Impl) EnqueueControllerOf(obj interface{}) {
 	}
 }
 
+// EnqueueLabelOf returns with an Enqueue func that takes a resource,
+// identifies its controller resource through given namespace and name labels,
+// converts it into a namespace/name string, and passes that to EnqueueKey.
+// Callers should pass in an empty string as namespace label key for obj
+// whose controller is of cluster-scoped resource.
+func (c *Impl) EnqueueLabelOf(namespaceLabel, nameLabel string) func(obj interface{}) {
+	return func(obj interface{}) {
+		object, err := meta.Accessor(obj)
+		if err != nil {
+			c.logger.Debug(zap.Error(err))
+			// To handle obj deletion, try to fetch info from DeletedFinalStateUnknown.
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				c.logger.Errorf("Couldn't get object from tombstone %#v", obj)
+				return
+			}
+			object, ok = tombstone.Obj.(metav1.Object)
+			if !ok {
+				c.logger.Errorf("The object that Tombstone contained is not of metav1.Object %#v", obj)
+				return
+			}
+		}
```

Comment:
Feels like we are going to want our own wrapper for the boilerplate up to this point.

Reply:
done